crates/turbo-tasks-fs/src/lib.rs

```diff
@@ -1403,10 +1408,15 @@ impl FileContent {
 
     pub fn parse_json(&self) -> FileJsonContent {
         match self {
-            FileContent::Content(file) => match serde_json::from_reader(file.read()) {
-                Ok(data) => FileJsonContent::Content(data),
-                Err(_) => FileJsonContent::Unparseable,
-            },
+            FileContent::Content(file) => {
+                let de = &mut serde_json::Deserializer::from_reader(file.read());
+                match serde_path_to_error::deserialize(de) {
+                    Ok(data) => FileJsonContent::Content(data),
+                    Err(e) => FileJsonContent::Unparseable(
+                        box UnparseableJson::from_serde_path_to_error(e),
+                    ),
+                }
```

Comment:
Should this be using `parse_json_rope_with_source_context`?

Reply:
No, since we don't want it wrapped in an anyhow error here, but return a structured error.